### PR TITLE
Feature/23415 feedback function persist to pupil event table storage

### PR DESCRIPTION
**Pull request by jon-shipley**
* Saves the pupil feedback message to the Admin DB - with one lookup to get the FK
* New FK `checkFormAllocation_id` added to `pupilFeedback` table that references the `checkFormAllocation` table rather than the `check` table which is deprecated.  This allows feedback to be provided by both old and new (HPA) systems.

### CODE REVIEW

File: functions/feedback/index.js [hunk 1]
```diff
@@ -1,19 +1,75 @@
...
+const sqlService = require('less-tedious')
+const { TYPES } = require('tedious')
+
+const sqlUtil = require('../lib/sql-helper')
+const schema = '[mtc_admin]'
+const feedbackTable = '[pupilFeedback]'
+
+module.exports = async function (context, feedbackMessage) {
+  if (typeof feedbackMessage !== 'object') {
+    throw new Error('Badly formed message')
```
Comment (GuyHarwood) [inline functions/feedback/index.js]: 🎖

File: functions/feedback/sample.data.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (GuyHarwood) [inline functions/feedback/sample.data.json]: this is a good idea 💯 
we should do this across all of them

Comment (GuyHarwood) [reply functions/feedback/sample.data.json]: card added to board for above.

File: functions/lib/sql-helper.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+const sqlService = require('less-tedious')
+
+const checkFormAllocationTable = '[checkFormAllocation]'
+const { TYPES } = require('tedious')
+const schema = '[mtc_admin]'
+
+/**
+ * Retrieve the checkFormAllocation data from the db
+ * @param checkCode
+ * @return {Promise<void>}
```
Comment (GeorgeChatzigiannis) [inline functions/lib/sql-helper.js]: @jon-shipley JSdoc return type is wrong.
You are returning an object(R.head) as far as I see.

**Pull request merged**